---
title: "Lecture 14: Transcriptomics and RNA-Seq data analysis"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## DESeq2 setup

We will use the DESeq2 package from the bioconductor. To install this we ran **in the CONSOLE** - note eval=FALSE option:

```{r eval=FALSE}
install.packages("BiocManager")
BiocManager::install()
BiocManager::install("DESeq2")
```

Read the dataset - the *countData* and *colData* files
```{r}
counts <- read.csv("Data/airway_scaledcounts.csv", stringsAsFactors = FALSE)
metadata <-  read.csv("Data/airway_metadata.csv", stringsAsFactors = FALSE)
```

Look at each
```{r}
head(counts)

head(metadata)
```

How many genes in this dataset?
```{r}
nrow(counts)
```

How many experiments do we have?
```{r}
ncol(counts)

#Can also look at metadata
nrow(metadata)
```


## Differential Gene Expression Analysis 
Lets perform some exploratory differential gene expression analysis. **Note:** Demonstration ***only,*** never do differential expression analysis like this.

We want to know if there is a difference in expression values for control (non-drug) vs treated (i.e. drug added cell lines).

First step is to find which experiments were the control experiments and then get the avergae values across all control experiments. Then we will do the same for drug treated.

Look at the entire object using `View()`
```{r}
View(metadata)
```

Calculate mean counts per gene across these samples
```{r}
control <- metadata[metadata[,"dex"]=="control",]

rowSums( counts[ ,control$id] )/4 
```

Calculate the mean values across the control columns of **countdata**
```{r}
control.mean <- rowSums(counts[,control$id])/length(control$id)

names(control.mean) <- counts$ensgene

control.mean
```

Do the same thing for the treated
```{r}
treated <- metadata[metadata[, "dex"]=="treated",]

treated.mean <- rowSums(counts[,treated$id])/length(treated$id)

names(treated.mean) <- counts$ensgene

treated.mean
```

Combine the meancount data
```{r}
meancounts <- data.frame(control.mean, treated.mean)

colSums(meancounts)
```

Plot control vs treated
```{r}
plot(control.mean, treated.mean, log = "xy", xlab = "Control", ylab = "Treated")
```

We can find candidate differentially expressed genes by looking for genes with a large change between control and dex-treated samples. We usually look at the log2 of the fold change, because this has better mathematical properties.

Here we calculate log2foldchange, add it to our meancounts data.frame and inspect the results either with the head() or the View() function for example.

```{r}
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

Filter out data to remove these genes
```{r}
#Look at meancounts columns from 1:2, which equal 0, the argument arr.ind shows exactly which ones have a value of zero
zero.vals <- which(meancounts[,1:2]==0, arr.ind=TRUE)

#Remove the duplicate elements in the first column of zero values
to.rm <- unique(zero.vals[,1])

#Find the meancounts of to.rm
mycounts <- meancounts[-to.rm,]

#Look at it
head(mycounts)
```

Let's find out how many counts are above/below 2
```{r}
up.ind <- mycounts$log2fc > 2
down.ind <- mycounts$log2fc < (-2)

sum(up.ind)
sum(down.ind)
```

```{r}
head(mycounts[up.ind,])
```

## Adding Annotation Data
We can add annotation from a supplied CSV file, such as those available from ENSEMBLE or UCSC. The annotables_grch38.csv annotation table links the unambiguous Ensembl gene ID to other useful annotation like the gene symbol, full gene name, location, Entrez gene ID, etc.

```{r}
anno <- read.csv("Data/annotables_grch38.csv")
head(anno)
```

## DESeq2 Analysis
DESeq2 is an R package for analyzing count-based NGS data like RNA-seq. It is available from Bioconductor. Bioconductor is a project to provide tools for analyzing high-throughput genomic data including RNA-seq, ChIP-seq and arrays.

```{r}
library(DESeq2)
citation("DESeq2")
```

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds
```

Run DESeq2
```{r}
dds <- DESeq(dds)
```

Get the results
```{r}
res <- results(dds)
res
```

## Volcano Plot

This figure will combine both Fold Change and the p-value into one overview figure indicating the proportion of genes that are both significantly regulated and display a high fold change.

```{r}
res$sig <- res$padj<0.05 & abs(res$log2FoldChange)>2

# How many of each?
table(res$sig)
```

```{r}
sum(is.na(res$sig))
```

```{r}
# Set the color palette for our plot
palette( c("gray","blue") )

plot( res$log2FoldChange,  -log(res$padj), 
 col=res$sig+1, ylab="-Log(P-value)", xlab="Log2(FoldChange)")

# Add some cut-off lines
abline(v=c(-2,2), col="darkgray", lty=2)
abline(h=-log(0.1), col="darkgray", lty=2)
```

```{r}
# Reset the color palette
palette("default")   
```

```{r}
# Setup our custom point color vector 
mycols <- rep("gray", nrow(res))
mycols[ abs(res$log2FoldChange) > 2 ]  <- "red" 

inds <- (res$padj < 0.01) & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"

#Volcano plot with custom colors 
plot( res$log2FoldChange,  -log(res$padj), 
 col=mycols, ylab="-Log(P-value)", xlab="Log2(FoldChange)" )

abline(v=c(-2,2), col="gray", lty=2)
abline(h=-log(0.1), col="gray", lty=2)
```

```{r}
library(ggplot2)

ggplot(as.data.frame(res), aes(log2FoldChange, -log10(pvalue), col=sig)) + 
    geom_point() + 
    ggtitle("Volcano plot")
```

# Save Results

```{r}
write.csv(res, file = "expression_results.csv")
```

